### PR TITLE
Cleanup TODOs in code related to multichannel

### PR DESCRIPTION
**Pull request by d-wysocki**
Some cleanup related to multichannel changes. Including:
- remove TODO in fixtures
- remove `skip` in unitests
- remove `FIXME` - refactor variant_resolver in `OrderLine` and `CheckoutLine`

### CODE REVIEW

**Review by IKarbowiak — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/product/tests/test_product.py [hunk 1]
```diff
@@ -1833,15 +1833,11 @@ def test_filter_products_by_wrong_attributes(user_api_client, product, channel_U
...
 def test_sort_products(user_api_client, product, channel_USD):
-    # set price and update date of the first product
-    product.minimal_variant_price_amount = 10
     product.updated_at = datetime.utcnow()
     product.save()
 
-    # Create the second product with higher price and date
     product.pk = None
     product.slug = "second-product"
-    product.minimal_variant_amount = 20
```
Comment (maarcingebala) [inline saleor/graphql/product/tests/test_product.py:1844]: Shouldn't this test check sorting by the discounted price? I think that's how the price sorter works.

File: saleor/graphql/product/resolvers.py [hunk 2]
```diff
@@ -141,3 +154,41 @@ def resolve_report_product_sales(period, channel_slug) -> ChannelQsContext:
...
+            requestor_has_access_to_all = models.Product.objects.user_has_access_to_all(
+                requester
+            )
+            if visible_in_listings or requestor_has_access_to_all:
+                return ChannelContext(node=variant, channel_slug=channel.slug)
+            return None
+
+        return (
+            ProductChannelListingByProductIdAndChannelSlugLoader(context)
+            .load((variant.product_id, channel))
```
Comment (fowczarek) [inline saleor/graphql/product/resolvers.py:187]:
```suggestion
            .load((variant.product_id, channel.slug))
```

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/product/tests/test_product.py (hunk 1 above)
Comment (d-wysocki) [reply saleor/graphql/product/tests/test_product.py:1844]: `PRICE` is sorting by price amount in ProductVarianChannelListing, we have other sorter to sort by discounted price - `MINIMAL_PRICE`

Comment (maarcingebala) [reply saleor/graphql/product/tests/test_product.py:1844]: OK I wasn't sure about this, thanks.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**